app/src/main/java/org/cru/godtools/fragment/ToolDetailsFragment.java

```diff
@@ -280,24 +288,23 @@ private void updateViews() {
             mTitle.setText(ModelUtils.getTranslationName(getContext(), mLatestTranslation, mTool));
         }
         bindShares(mShares, mTool);
-        if (mDescription != null) {
-            mDescription.setText(ModelUtils.getTranslationDescription(getContext(), mLatestTranslation, mTool));
-        }
-        if (mLanguagesHeader != null) {
-            final int count = mLanguages.size();
-            mLanguagesHeader.setText(mLanguagesHeader.getResources()
-                                             .getQuantityString(R.plurals.label_tools_languages, count, count));
-        }
-        if (mLanguagesView != null) {
-            mLanguagesView.setVisibility(mLanguages.isEmpty() ? View.GONE : View.VISIBLE);
-            mLanguagesView.setText(Stream.of(mLanguages)
-                                           .map(l -> LocaleUtils.getDisplayName(l, mLanguagesView.getContext(), null,
-                                                                                null))
-                                           .withoutNulls()
-                                           .sorted(String.CASE_INSENSITIVE_ORDER)
-                                           .reduce((l1, l2) -> l1 + ", " + l2)
-                                           .orElse(""));
+
+        final int count = mLanguages.size();
+        if (mViewPager != null) {
+            ToolsDetailAdapter adapter = new ToolsDetailAdapter();
+            mViewPager.setAdapter(adapter);
+            if (mTabLayout != null) {
+                mTabLayout.setupWithViewPager(mViewPager, true);
+                if (mTabLayout.getTabAt(1) != null) {
+                    mTabLayout.getTabAt(1).setText(requireContext().getResources()
+                            .getQuantityString(R.plurals.label_tools_languages, count, count));
+                }
+                if (mTabLayout.getTabAt(0) != null) {
+                    mTabLayout.getTabAt(0).setText(R.string.label_tools_about);
```

Comment:
why can't you just set this text directly in the XML?

Reply:
<img width="417" alt="Android_Emulator_-_Pixel_3a_API_29_5554" src="https://user-images.githubusercontent.com/970048/68155302-709a4580-ff17-11e9-88d5-0a243fa251f0.png">

This is what I get when I try to set the text in the XML.

Reply:
You should leverage `PagerAdapter.getTitle()` to drive the tab titles, it's what the method is actually for

Reply:
@gyasistory did you look into using the `getPageTitle` method as Daniel suggested?

Reply:
Looks like you did, but just making sure.